src/System.Runtime/tests/System/StringGetHashCodeTests.cs

```diff
@@ -0,0 +1,60 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using Xunit;
+
+namespace System.Tests
+{
+    public class StringGetHashCodeTests : RemoteExecutorTestBase
+    {
+        [Fact]
+        public void GetHashCode_UseSameStringInTwoProcesses_ReturnsDifferentHashCodes()
+        {
+            EnsureHashCodeDifferentForChildString("abc".GetHashCode().ToString());
+        }
+
+        private static void EnsureHashCodeDifferentForChildString(string origHashCode, string timesInvoked = "0")
```

Comment:
I think it would be clearer to inline this in the test itself. There is no need to create a new method to recurse (interesting that it works to RemoteInvoke transitively...) when a simple do/while loop would work and be clearer.

To pass back the information that it failed, you can use the exit code. Expose ExitCode as a property on RemoteInvokeHandle and read it off after the process quits. (Set RemoteInvokeOptions.CheckExitCode = false, so it doesn't fail the test on an unexpected error code)

Reply:
Or simpler still you might be able to pass the hashcode back as the exit code! Making the remote process a one liner.